autofit/optimize/non_linear/samples.py

```diff
@@ -21,39 +22,111 @@ def __init__(self, model, paths):
 
     @property
-    def most_likely_instance(self) -> model.ModelInstance:
-        """ The best-fit model sampled by the non-linear search (corresponding to the maximum log-likelihood), returned
-        as a model instance."""
-        return self.model.instance_from_vector(vector=self.most_likely_vector)
+    def max_log_likelihood_instance(self) -> model.ModelInstance:
+        """  The parameters of the maximum log likelihood sample of the non-linear search returned as a model instance."""
+        return self.model.instance_from_vector(vector=self.max_log_likelihood_vector)
+
+    @property
+    def max_log_posterior_index(self) -> int:
+        """The index of the sample with the highest log posterior."""
+        return int(np.argmax(self.log_posteriors))
+
+    @property
+    def max_log_posterior_vector(self) -> [float]:
+        """ The parameters of the maximum log posterior sample of the non-linear search returned as a list of values."""
+        return self.parameters[self.max_log_posterior_index]
+
+    @property
+    def max_log_posterior_instance(self) -> model.ModelInstance:
+        """  The parameters of the maximum log posterior sample of the non-linear search returned as a model instance."""
+        return self.model.instance_from_vector(vector=self.max_log_posterior_vector)
+
+    @property
+    def pdf_converged(self) -> bool:
+        """ To analyse and visualize chains using *GetDist*, the analysis must be sufficiently converged to produce
+        smooth enough PDF for analysis. This property checks whether the non-linear search's chains are sufficiently
+        converged for *GetDist* use.
+
+        For *Dynesty*, during initial sampling one accepted live point typically has > 99% of the probabilty as its
+        log_likelihood is significantly higher than all other points. Convergence is only achieved late in sampling when
+        all live points have similar log_likelihood and sampling probabilities."""
+        try:
+            densities_1d = list(
+                map(lambda p: self.pdf.get1DDensity(p), self.pdf.getParamNames().names)
+            )
+
+            if densities_1d == []:
+                return False
+
+            return True
+        except Exception:
+            return False
+
+    @property
+    def pdf(self):
+        """An interface to *GetDist* which can be used for analysing and visualizing the non-linear search chains.
+
+        *GetDist* can only be used when chains are converged enough to provide a smooth PDF and this convergence is
+        checked using the *pdf_converged* bool before *GetDist* is called.
+
+        https://github.com/cmbant/getdist
+        https://getdist.readthedocs.io/en/latest/
+
+        For *Dynesty*, chains are passed to *GetDist* using the pickled sapler instance, which contains the physical
+        model parameters of every accepted sample, their likelihoods and weights.
+        """
+        import getdist
```

Comment:
Did we say we'd get rid of getdist? It's a pretty shitty library. Did we import it locally because it takes ages to initialise?

Reply:
We're still pretty reliant on GetDist, and its the best thing out there for the visualization once you get through the awkward interface. Im not sure why we import it locally, it goes back to something from a long time ago which may or may not be relevent anymore. I'll try moving the import up when I do some integration tests.